src/biguint.rs

```diff
@@ -1026,6 +1026,52 @@ impl Integer for BigUint {
     }
 }
 
+impl Roots for BigUint {
+    fn nth_root(&self, n: u32) -> Self {
```

Comment:
I think it would probably be valuable to have dedicated `sqrt` and `cbrt` implementations -- did you try it?  Maybe benchmarks would prove me wrong, but we can also wait until later.

Reply:
Here are the benchmark results:
```bash
test roots_cbrt            ... bench:      48,670 ns/iter (+/- 1,322)
test roots_cbrt_nth_root_3 ... bench:      52,285 ns/iter (+/- 1,541)
test roots_sqrt            ... bench:      53,752 ns/iter (+/- 4,954)
test roots_sqrt_nth_root_2 ... bench:      57,622 ns/iter (+/- 1,585)
```
`roots_cbrt` and `roots_sqrt` test the optimized versions - I took `sqrt` from #51 and adjusted it for `cbrt`. The other two benchmarks just called `nth_root(2)` and `nth_root(3)` - of course, for the sake of benchmark, I didn't forward the calls to `nth_root` with `n=2` and `n=3` to the optimized `sqrt` and `cbrt`, but I'll definitely do that, since I see this is how it's done in impls for primitive types as well.

I would prefer to incorporate the dedicated implementations in this PR, although I'm struggling a little since I'm not particularly fond of having  parts of `nth_root`, `sqrt`, `cbrt` replicated in all the methods. But I'm not too sure what (if anything) would be best to do about it :slightly_smiling_face: